script/pgsqlms

```diff
@@ -718,21 +721,23 @@ sub _check_locations {
         }
 
         # Remove this node from the known nodes list.
-        $partition_nodes =~ s/(?:^|\s)$row->[0](?:\s|$)/ /g;
+        $list_nodes =~ s/(?:^|\s)$row->[0](?:\s|$)/ /g;
     }
 
-    $partition_nodes =~ s/(?:^\s+)|(?:\s+$)//g;
+    $list_nodes =~ s/(?:^\s+)|(?:\s+$)//g;
 
-    # If there are still nodes in "partition_nodes", it means there is no
+    # If there are still nodes in "list_nodes", it means there is no
     # corresponding line in "pg_stat_replication".
-    foreach my $node (split /\s+/ => $partition_nodes) {
+    # In that case, we set the master_score to -1000 if it exists already.
+    foreach my $node (split /\s+/ => $list_nodes) {
         # Exclude the current node.
         next if $node eq $nodename;
 
         ocf_log( 'warning', sprintf
             '_check_locations: "%s" is not connected to the primary, set score to -1000',
             $node );
-        _set_master_score( -1000, $node );
+        $node_score = _get_master_score( $node );
+        _set_master_score( -1000, $node ) if $node_score ne '';
```

Comment:
I'm wondering about this condition: `if $node_score ne ''`.

I guess it should be `if $node_score ne '-1000'` to not bother attrd with an useless update score?